kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/ALBookingRMAdapter.java

```diff
@@ -0,0 +1,522 @@
+            }
+
+            if (!TextUtils.isEmpty(hotel.getHotelPicture())) {
+                Glide.with(context).load(hotel.getHotelPicture()).into(holder.productImage);
+            } else {
+                holder.productImage.setImageDrawable(null);
+            }
+
+            holder.noOfGuestTv.setText(String.valueOf(hotel.getNoOfGuest()));
+
+            String text = context.getString(R.string.room_night_detail, hotel.getNoOfNights());
+            holder.totalPriceHeaderTv.setText(text);
+
+            holder.priceTv.setText(context.getString(R.string.rupee_symbol, String.valueOf(hotel.getPrice().getRoomPrice())));
+
+            holder.totalPriceTv.setText(context.getString(R.string.rupee_symbol, String.valueOf(hotel.getPrice().getRoomPrice() * hotel.getNoOfNights())));
+
+            holder.bookAction.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    hotel.setSessionId(model.getSessionId());
+                    listener.onAction(context, AlRichMessage.SEND_ROOM_DETAILS_MESSAGE, null, hotel, null);
+                }
+            });
+        }
+    }
+
+    private void bindBookingDetailHolder(final BookingDetailsHolder holder, final int position) {
+        final ALBookingDetailsModel detailsModel = new ALBookingDetailsModel();
+        detailsModel.setSessionId(model.getSessionId());
+
+        final ALBookingDetailsModel.ALBookingDetails bookingDetails = detailsModel.getPersonInfo();
+
+        holder.titleSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                bookingDetails.setTitle(titleList.get(position));
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+                bookingDetails.setTitle(context.getString(R.string.list_item_title));
+            }
+        });
+
+        holder.submitAction.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                if (TextUtils.isEmpty(holder.firstNameEt.getText().toString().trim())
+                        || TextUtils.isEmpty(holder.lastNameEt.getText().toString().trim())
+                        || TextUtils.isEmpty(holder.emailIdEt.getText().toString().trim())
+                        || TextUtils.isEmpty(holder.firstNameEt.getText().toString().trim())
+                        || "Title *".equals(titleList.get(holder.titleSpinner.getSelectedItemPosition()))) {
+                    Toast.makeText(context, R.string.mandatory_fields, Toast.LENGTH_SHORT).show();
+                } else {
+                    bookingDetails.setTitle(titleList.get(holder.titleSpinner.getSelectedItemPosition()));
+                    bookingDetails.setFirstName(holder.firstNameEt.getText().toString().trim());
```

Comment:
we can add trim inside the setter methods of bookingDetails, then we will not need to trim it everywhere

Reply:
done